### PR TITLE
discovery: Report origin of generated service name

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Report the origin of the generated service name.  See 
https://datadoghq.atlassian.net/wiki/spaces/UT/pages/4397760938/ for more information.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1351

### Describe how to test/QA your changes

Automated tests are included in the PR.

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/usm/service_test.go [hunk 1]
```diff
@@ -666,6 +669,10 @@ func TestExtractServiceMetadata(t *testing.T) {
 				require.Equal(t, tt.expectedGeneratedName, meta.Name)
 				require.Equal(t, tt.expectedAdditionalServices, meta.AdditionalNames)
 				require.Equal(t, tt.ddServiceInjected, meta.DDServiceInjected)
+				require.NotEmpty(t, meta.Source)
+				if len(tt.expectedSource) != 0 {
```
Comment (Yumasi) [inline pkg/collector/corechecks/servicediscovery/usm/service_test.go:673]: Why is the source checked only on some cases?

File: pkg/collector/corechecks/servicediscovery/usm/jee.go [hunk 2]
```diff
@@ -300,16 +300,16 @@ func (je jeeExtractor) doExtractContextRoots(extractor vendorExtractor, app *jee
 // extractServiceNamesForJEEServer takes args, cws and the fs (for testability reasons) and, after having determined the vendor,
 // If the vendor can be determined, it returns the context roots if found, otherwise the server name.
 // If the vendor is unknown, it returns a nil slice
```
Comment (Yumasi) [inline pkg/collector/corechecks/servicediscovery/usm/jee.go:302]: This comment needs updating with the added return value.

**Review by Yumasi — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/usm/service_test.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/usm/service_test.go]: Fixed.

File: pkg/collector/corechecks/servicediscovery/usm/jee.go (hunk 2 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/usm/jee.go]: Done.

**Pull request merged**